test/parallel/test-repl-unsafe-array-iteration.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn } = require('child_process');
+
+function run(input, expectation) {
+  const node = spawn(process.argv0);
+
+  node.stderr.on('data', common.mustNotCall());
+  node.on('exit', common.mustCall((code) => assert.strictEqual(code, 0)));
+
+  node.stdin.write(input);
+  node.stdin.end();
+}
+
+run('delete Array.prototype[Symbol.iterator];',
+    'TypeError: Found non-callable @@iterator');
+
+run('const ArrayIteratorPrototype =' +
+      'Object.getPrototypeOf(Array.prototype[Symbol.iterator]());' +
+      'delete ArrayIteratorPrototype.next;',
+    'TypeError: fn is not a function');
```

Comment:
We need to run `node --interactive` to get REPL, otherwise it evals `stdin` and doesn't run any code from `repl.js`. That may explain why you were getting error code `0` previously.

```suggestion
const replProcess = spawn(process.argv0, ['--interactive'], {
  stdio: ['pipe', 'pipe', 'inherit'],
  windowsHide: true,
});

replProcess.on('error', common.mustNotCall());

const replWaiter = (async function* () {
  let ready;
  const replReadyBuffer = Buffer.from('> ');
  function checkData(data) {
    ready = Buffer.compare(data, replReadyBuffer) === 0;
  }
  replProcess.stdout.on('data', checkData);

  const processCrashed = new Promise((resolve, reject) =>
    replProcess.on('exit', reject)
  );
  while (true) {
    await Promise.race([new Promise(setImmediate), processCrashed]);
    yield ready;
    ready = false;
  }
})();
async function waitForREPL() {
  while (!(await replWaiter.next()).value);
}

async function writeOnREPL(data, expectResult) {
  await waitForREPL();
  await new Promise((resolve, reject) => {
    replProcess.stdin.write(data, (err) => (err ? reject(err) : resolve()));
  });
  if (expectResult) {
    replProcess.stdout.once(
      'data',
      common.mustCall((data) =>
        assert.match(data.toString('utf8'), expectResult)
      )
    );
  }
}

async function main() {
  await writeOnREPL(
    'const ArrayIteratorPrototype =' +
      '  Object.getPrototypeOf(Array.prototype[Symbol.iterator]());\n'
  );
  await writeOnREPL('delete ArrayIteratorPrototype.next;\n');
  await writeOnREPL('delete Array.prototype[Symbol.iterator];\n');

  await writeOnREPL(
    'for(const x of [3, 2, 1]);\n',
    /Uncaught TypeError: \[3,2,1\] is not iterable/
  );
  await writeOnREPL('.exit\n');

  assert(!replProcess.connected);
}

main().then(common.mustCall());

```